myriad-scheduler/src/main/java/com/ebay/myriad/MyriadModule.java

```diff
@@ -95,11 +93,21 @@ protected void configure() {
 
     @Provides
     @Singleton
-    SchedulerState providesSchedulerState(MyriadConfiguration cfg,
-        State stateStore) {
-
+    SchedulerState providesSchedulerState(MyriadConfiguration cfg) {
         LOGGER.debug("Configuring SchedulerState provider");
-        MyriadState state = new MyriadState(stateStore);
-        return new SchedulerState(state);
+        MyriadStateStore myriadStateStore = null;
+        if (cfg.isHAEnabled()) {
+            myriadStateStore = providesMyriadStateStore();
```

Comment:
yes, agree. Will add the RuntimeException. However the code is ok with myriadStateStore being null (in case where HA is disabled)